db/app/app-tests/app.test.js

```diff
@@ -0,0 +1,46 @@
+const request = require("supertest");
+const seed = require("../../seeds/seed");
+const db = require("../../../db/connection");
+const {
+  articleData,
+  commentData,
+  topicData,
+  userData,
+} = require("../../data/test-data");
+require("jest-sorted");
+const { app } = require("../app");
+
+afterAll(() => {
+  db.end();
+});
+
+beforeEach(() => {
+  return seed({ articleData, commentData, topicData, userData });
+});
+
+describe("GET /api/topics", () => {
+  test("200: should return topic data with slug and description properties ", () => {
+    return request(app)
+      .get("/api/topics")
+      .expect(200)
+      .then(({body}) => {
+        const  {topics} = body
+        topics.forEach((topic) => {
+          expect(topic).toMatchObject({
+            description: expect.any(String),
+            slug: expect.any(String),
+          });
+        });
+      });
+  });
+  test("200: should return an array of all topics within the databse", () => {
+    return request(app)
+      .get("/api/topics")
+      .expect(200)
+      .then(({body}) => {
+        const {topics} = body
+        expect(topics).toHaveLength(3);
```

Comment:
You could and should run this array length assertion in the above test as we need to check the length of an array if using a foreEach to make assertions about the array's contents (we could get a false positive if the array was empty). This also mean you only need one test (with two assertions), not two tests. The assertion on line 43 is also covered by the toMatchObject matcher so can be omitted.